File: Matlab MRI/weighted.m
%{

This program plots the magnitudes and
contrast by weighting T1 and T2 values.

%}

Mo = 1;
TR = 0:1:5000;

% get T1 values
prompt1 = 'T1 for CSF: ';
csfT1 = input(prompt1);

prompt2 = 'T1 for Gray matter: ';
grayT1 = input(prompt2);

prompt3 = 'T1 for White matter: ';
whiteT1 = input(prompt3);

prompt4 = 'T1 for Fat: ';
fatT1 = input(prompt4);

% get T2 values
prompt5 = 'T2 for CSF: ';
csfT2 = input(prompt5);

prompt6 = 'T2 for Gray matter: ';
grayT2 = input(prompt6);

prompt7 = 'T2 for White matter: ';
whiteT2 = input(prompt7);

prompt8 = 'T2 for Fat: ';
fatT2 = input(prompt8);

% get TE value
prompt9 = 'TE: ';
TE = input(prompt9);


% equations for magnitude with each T1 and T2 value
CSF = Mo .*(1-exp(-TR/csfT1)).* exp(-TE/csfT2);
gray = Mo .* (1-exp(-TR/grayT1)).* exp(-TE/grayT2);
white = Mo .* (1-exp(-TR/whiteT1)).* exp(-TE/whiteT2);
fat = Mo .*(1-exp(-TR/fatT1)).* exp(-TE/fatT2);


figure()

% plot magnitude
subplot(221)
plot(TR, CSF);
hold on 
plot(TR, gray);
plot(TR, white);
plot(TR, fat);
hold off
title('Weighted Imaging');
grid on 

% plot contrast between gray matter and csf
subplot(222)
csfcontrast = gray - CSF;
plot(TR, csfcontrast)
title('Contrast Between Gray Matter and CSF')
hold on
if csfcontrast<=0
    [M,I] = min(csfcontrast); 
else 
    [M,I] = max(csfcontrast) ;
end
plot(I, M, 'o');
hold off
grid on 

% plot contrast between gray matter and white matter
subplot(223)
whitecontrast = gray - white;
plot(TR, whitecontrast)
hold on
if whitecontrast<=0
    [M,I] = min(whitecontrast); 
else 
    [M,I] = max(whitecontrast) ;
end
plot(I, M, 'o');
hold off
title('Contrast Between Gray and White Matter')
grid on 

% plot contrast between gray matter and fat
subplot(224)
fatcontrast = gray - fat;
plot(TR, fatcontrast)
hold on
if fatcontrast<=0
    [M,I] = min(fatcontrast); 
else 
    [M,I] = max(fatcontrast) ;
end
plot(I, M, 'o');
hold off
title('Contrast Between Gray Matter and Fat')
grid on 

File: Python MRI/Weighted.py
"""

This program plots the magnitudes and
contrast by weighting T1 and T2 values.

"""


import numpy as np
import matplotlib.pyplot as plt

TR = np.arange(0,5000)

Mo = 1

# get T1 values
csfT1 = int(input('T1 for CSF: '))
grayT1 = int(input('T1 for gray matter: ' ))
whiteT1 = int(input('T1 for white matter: '))
fatT1 = int(input('T1 for fat: '))

# get T2 values
csfT2 = int(input('T2 for CSF: '))
grayT2 = int(input('T2 for gray matter: ' ))
whiteT2 = int(input('T2 for white matter: '))
fatT2 = int(input('T2 for fat: '))

# get TE value
TE = int(input('TE: ' ))

# equations for signal magnitude
csf = Mo*(1-np.exp(-TR/csfT1))*(np.exp(-TE/csfT2))
gray = Mo*(1-np.exp(-TR/grayT1))*(np.exp(-TE/grayT2))
white = Mo*(1-np.exp(-TR/whiteT1))*(np.exp(-TE/whiteT2))
fat = Mo*(1-np.exp(-TR/fatT1))*(np.exp(-TE/fatT2))

# plot magnitudes
plt.subplot(221)
plt.plot(TR, csf, label = 'CSF')
plt.plot(TR, gray, label = 'Gray matter')
plt.plot(TR, white, label = 'White matter')
plt.plot(TR, fat, label = 'Fat')
plt.title('Weighted Imaging')
plt.xlabel('TR (msecs)')
plt.ylabel('Magnitude')
plt.grid(True)
plt.legend()
plt.xlim(0, 5000)
plt.ylim(ymin = 0)

# plot contrast between gray matter and CSF
plt.subplot(222)
csfcontrast = gray - csf
plt.plot(TR, csfcontrast)
if np.all(csfcontrast)<=0:
    plt.plot(np.argmin(csfcontrast), np.min(csfcontrast), marker = 'o')
else:
    plt.plot(np.argmax(csfcontrast), np.max(csfcontrast), marker = 'o') 
plt.xlim(0, 5000)
plt.grid(True)
plt.xlabel('TR (msecs)')
plt.ylabel('Contrast')
plt.title('Contrast Between Gray Matter and CSF')

# plot contrast between gray matter and white matter
plt.subplot(223)
whitecontrast = gray - white
plt.plot(TR, whitecontrast)
if np.all(whitecontrast <= 0):
    plt.plot(np.argmin(whitecontrast), np.min(whitecontrast), marker = 'o')
else:
    plt.plot(np.argmax(whitecontrast), np.max(whitecontrast), marker = 'o')
plt.xlim(0, 5000)
plt.grid(True)
plt.xlabel('TR (msecs)')
plt.ylabel('Contrast')
plt.title('Contrast Between Gray and White Matter')

#plot contrast between gray matter and fat
plt.subplot(224)
fatcontrast = gray - fat
plt.plot(TR, fatcontrast)
if np.all(fatcontrast<=0):
    plt.plot(np.argmin(fatcontrast), np.min(fatcontrast), marker = 'o')
else:
    plt.plot(np.argmax(fatcontrast), np.max(fatcontrast), marker = 'o') 
plt.xlim(0, 5000)
plt.grid(True)
plt.xlabel('TR (msecs)')
plt.ylabel('Contrast')
plt.title('Contrast Between Gray Matter and Fat')


#show graphs
plt.tight_layout()
plt.show()



